dashboard/lib/tasks/asset_sync.rake

```diff
@@ -54,7 +54,7 @@ Rerun `assets:precompile` to regenerate new assets and try again."
   # updated if there are any changes to Sprockets processors.
   module NoDoubleDigest
     def digest_path
-      logical_path.match?(/wp\h{32}/) ? logical_path : super
+      logical_path.match?(/wp\h{20}/) ? logical_path : super
```

Comment:
I'm confused - does this logic depend on Webpack generating exactly 20-character hashes?  How does this interact with the 32-character hashes we drop after copy-webpack-plugin, above?

Reply:
Sorry, I glossed over something here. The regex on this line matches either a 32- or 20-character hash, since all it checks for is `wp` followed by at least 20 hex characters. I will add a comment stating this.

The `[contenthash]` in webpack's [output.filename](https://webpack.js.org/configuration/output/#outputfilename) results in a 20-character hash, which is why this line had to be changed: https://github.com/code-dot-org/code-dot-org/blob/a4b151060a2ad3ff12c01bd00fbe9b4d1ee8e6c8/apps/webpack.js#L329-L335

strangely, the `[contenthash]` placeholder consumed by `copy-webpack-plugin` gives a 32-character hash: https://github.com/code-dot-org/code-dot-org/blob/a4b151060a2ad3ff12c01bd00fbe9b4d1ee8e6c8/apps/Gruntfile.js#L886-L893

Reply:
Could these references be added to the comment so this knowledge doesn't get lost in the future? Without this reference I don't think it'll be obvious why it's a '20 or 32 character string' rather than one or the other.

Reply:
will do.